### PR TITLE
Add documentation for input validation

### PR DESCRIPTION
**Pull request by RebeccaAnne**
Added a documentation page for Input Validation. Covers the issues in https://github.com/microsoft/AdaptiveCards/issues/4391.

### CODE REVIEW

File: AdaptiveCards/authoring-cards/input-validation.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+## Fields to be validated and submitted
+
+Inputs will be validated when the user clicks on an Action.Submit action in the card. Those inputs which will be validated and submitted for a given Action.Submit action are:
+
+ - Inputs on the same card as the Action.Submit
+ - Inputs on any parent cards of the card containing the Action.Submit in the case of a card under an Action.ShowCard
+
+If those inputs pass validation, the values in their fields will be passed back to the client. If they do not pass validation, the error messages for the invalid inputs will be shown, and the submit will not be sent.
+
+> [!NOTE]
```
Comment (shalinijoshi19) [inline AdaptiveCards/authoring-cards/input-validation.md:41]: Is this markdown syntax? The Note warning seems to render funny in Preview?

File: AdaptiveCards/authoring-cards/input-validation.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+title: Input Validation
+author: rebeccaanne
+ms.author: rebecch
+ms.date: 07/24/2020
+ms.topic: article
+---
+
+# Input Validation
+
+Versions 1.3 and later of the Adaptive Cards schema support client side input validation of Input types.
```
Comment (shalinijoshi19) [inline AdaptiveCards/authoring-cards/input-validation.md:11]: Nit nit:  ',' after schema

File: AdaptiveCards/authoring-cards/input-validation.md (hunk 1 above)
Comment (RebeccaAnne) [reply AdaptiveCards/authoring-cards/input-validation.md:41]: I think it must be a thing specific to docs? I copied it from one of the existing pages. To see how it looks in real life check out this: https://review.docs.microsoft.com/en-us/adaptive-cards/authoring-cards/input-validation?branch=rebecch%2FinputValidationDocs

File: AdaptiveCards/authoring-cards/input-validation.md (hunk 2 above)
Comment (RebeccaAnne) [reply AdaptiveCards/authoring-cards/input-validation.md:11]: I'm not sure a comma goes there? I could do: "Adaptive cards, in versions 1.3 and later of the schema, supports client side input validation of Input types" Does that seem clearer?

Comment (shalinijoshi19) [reply AdaptiveCards/authoring-cards/input-validation.md:11]: Should we include a note about guidance and expectations for custom input types (backcompat and newer ones)?

Comment (shalinijoshi19) [reply AdaptiveCards/authoring-cards/input-validation.md:11]: > I'm not sure a comma goes there? I could do: "Adaptive cards, in versions 1.3 and later of the schema, supports client side input validation of Input types" Does that seem clearer?

Yeah or maybe even "In versions 1.3 and later of the schema, AdaptiveCards supports ... "

**Comment by shalinijoshi19** [comment]
@RebeccaAnne this would then need to be CP'd into live as well IIRC

File: AdaptiveCards/authoring-cards/input-validation.md (hunk 2 above)
Comment (RebeccaAnne) [reply AdaptiveCards/authoring-cards/input-validation.md]: > Should we include a note about guidance and expectations for custom input types (backcompat and newer ones)?

Is there something specific you had in mind here?

Comment (shalinijoshi19) [reply AdaptiveCards/authoring-cards/input-validation.md]: In terms of the Extensibility options [here](https://docs.microsoft.com/en-us/adaptive-cards/rendering-cards/extensibility) do things just work as far as any custom inputs they might have had, post-upgrade to 1.3/renderer lib 2.0?

Comment (RebeccaAnne) [reply AdaptiveCards/authoring-cards/input-validation.md]: Ok, digging in a bit I have some questions myself. I think that's going to have to be beyond the scope of this change. The extensibility documentation in general and for inputs in particular even prior to this work is pretty lacking.  I'm going to log a new issue for input extensibility documentation to follow up.

Comment (RebeccaAnne) [reply AdaptiveCards/authoring-cards/input-validation.md]: #319

File: AdaptiveCards/toc.yml [hunk 3]
```diff
@@ -18,6 +18,8 @@
     href: authoring-cards/speech.md
```
Comment (ame1234) [inline AdaptiveCards/toc.yml:18]: Arlyss engebretson

File: AdaptiveCards/toc.yml [hunk 4]
```diff
@@ -18,6 +18,8 @@
...
   - name: Text Features
     href: authoring-cards/text-features.md
+  - name: Input Validation
+    href: authoring-cards/input-validation.md
 - name: Rendering Cards
   items:
   - name: Getting Started
```
Comment (ame1234) [inline AdaptiveCards/toc.yml:23]: Arlyss Engebretson

**Pull request merged**